src/index.ts

```diff
@@ -231,6 +233,70 @@ function launchMetals(
       );
     }
 
+    registerCommand("metals.new-scala-file", async () => {
+      const currentDoc = await workspace.document;
```

Comment:
can this fail?

Reply:
As far as I understand it, no. Once you open vim you have a workspace. Even if you open directly with no file into a directory, you'll simply get either nerdTree if you have it installed or else just netwr. _However_, not that you bring it up, I thought of it when I first started, but then didn't test it. I'll need to check what happens if you issue this from there.

Reply:
Ok, checked. It just puts it in the root, which is what I sort of expected. So no, I don't see this ever failing.